router/batchrouter/asyncdestinationmanager/yandexmetrica/yandexmetrica.go

```diff
@@ -0,0 +1,382 @@
+		}
+		return IdStruct{id: ym.ClientID.(string), clientType: idClientMap["ClientId"], headerName: "ClientId"}, nil
+	case ym.YclID != nil:
+		_, ok := ym.YclID.(string)
+		if !ok {
+			return IdStruct{}, fmt.Errorf("non-string data for Yclid is not supported")
+		}
+		return IdStruct{id: ym.YclID.(string), clientType: idClientMap["Yclid"], headerName: "Yclid"}, nil
+	case ym.UserID != nil:
+		_, ok := ym.UserID.(string)
+		if !ok {
+			return IdStruct{}, fmt.Errorf("non-string data for UserId is not supported")
+		}
+		return IdStruct{id: ym.UserID.(string), clientType: idClientMap["UserId"], headerName: "UserId"}, nil
+	default:
+		return IdStruct{}, fmt.Errorf("no valid id found in message object")
+	}
+}
+
+type YandexMetricaBulkUploader struct {
+	logger          logger.Logger
+	Client          *http.Client
+	destinationInfo *oauthv2.DestinationInfo
+}
+
+func NewManager(destination *backendconfig.DestinationT, backendConfig backendconfig.BackendConfig) (*YandexMetricaBulkUploader, error) {
+	destinationInfo := &oauthv2.DestinationInfo{
+		Config:           destination.Config,
+		DefinitionConfig: destination.DestinationDefinition.Config,
+		WorkspaceID:      destination.WorkspaceID,
+		DefinitionName:   destination.DestinationDefinition.Name,
+		ID:               destination.ID,
+	}
+	yandexUploadManager := &YandexMetricaBulkUploader{
+		destinationInfo: destinationInfo,
+		logger:          logger.NewLogger().Child("batchRouter").Child("AsyncDestinationManager").Child("YandexMetrica").Child("YandexMetricaBulkUploader"),
+	}
+	cache := oauthv2.NewCache()
+	optionalArgs := &oauthv2httpclient.HttpClientOptionalArgs{
+		Logger:    yandexUploadManager.logger,
+		Augmenter: augmenter.YandexReqAugmenter,
+	}
+	originalHttpClient := &http.Client{Transport: &http.Transport{}}
+	// This client is used for uploading data to yandex metrica
+	yandexUploadManager.Client = oauthv2httpclient.NewOAuthHttpClient(
+		originalHttpClient,
+		oauthv2common.RudderFlowDelivery,
+		&cache,
+		backendConfig,
+		augmenter.GetAuthErrorCategoryForYandex,
+		optionalArgs,
+	)
+	return yandexUploadManager, nil
+}
+
+// return a success response for the poll request every time by default
+func (ym *YandexMetricaBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
+	return common.PollStatusResponse{}
+}
+
+// return a success response for the getUploadStats request every time by default
+func (ym *YandexMetricaBulkUploader) GetUploadStats(UploadStatsInput common.GetUploadStatsInput) common.GetUploadStatsResponse {
+	return common.GetUploadStatsResponse{}
+}
+
+func generateCSVFromJSON(jsonData []byte, goalId string) (string, string, error) {
+	// Define an empty map to store the parsed JSON data
+	var ymMsgs []YandexMetricaMessage
+	inputData := gjson.GetBytes(jsonData, "input").String()
+	err := json.Unmarshal([]byte(inputData), &ymMsgs)
+	if err != nil {
+		return "", "", fmt.Errorf("error while unmarshalling transformed response: %v", err)
```

Comment:
[minor] `error while` is superfluous
```suggestion
		return "", "", fmt.Errorf("unmarshalling transformed response: %v", err)
```

Reply:
Updated